### PR TITLE
es_visitor: support for legacy jhep/jcap queries

### PR DESCRIPTION
**Pull request by MJedr**
* INSPIR-3400

### CODE REVIEW

File: inspire_query_parser/visitors/elastic_search_visitor.py [hunk 1]
```diff
@@ -595,6 +607,18 @@ def _generate_journal_nested_queries(self, value):
             wrap_queries_in_bool_clauses_if_more_than_one(queries_for_each_field, use_must_clause=True)
         )
 
+    def _check_if_nested_field_in_fieldnames_list(self, fieldnames):
```
Comment (monaawi) [inline inspire_query_parser/visitors/elastic_search_visitor.py:610]: leftover?

File: tests/test_elastic_search_visitor.py [hunk 2]
```diff
@@ -2600,3 +2600,101 @@ def test_nested_query_exact_last_name():
 
     generated_es_query = _parse_query(query_string)
     assert generated_es_query == expected_es_query
+
+
+def test_elastic_search_visitor_type_code_legacy_compatible_jhep_search():
```
Comment (monaawi) [inline tests/test_elastic_search_visitor.py:2605]: I would rename this to ```test_elastic_search_visitor_find_journal_with_year```

File: tests/test_elastic_search_visitor.py [hunk 3]
```diff
@@ -2600,3 +2600,101 @@ def test_nested_query_exact_last_name():
...
+             }
+          }
+       }
+    }
+
+    generated_es_query = _parse_query(query_str)
+    assert generated_es_query == expected_es_query
+
+
+def test_elastic_search_visitor_type_code_legacy_compatible_jcap_search():
```
Comment (monaawi) [inline tests/test_elastic_search_visitor.py:2654]: I would remove this test as it isn't adding any value. We already have test that it uses the proper journal title, you just needed to test that the year is added too

File: tests/test_elastic_search_visitor.py [hunk 4]
```diff
@@ -2600,3 +2600,52 @@ def test_nested_query_exact_last_name():
 
     generated_es_query = _parse_query(query_string)
     assert generated_es_query == expected_es_query
+
+
+def test_elastic_search_visitor_find_journal_with_year():
+    query_str = "j jhep,0903,112,2009"
```
Comment (michamos) [inline tests/test_elastic_search_visitor.py:2606]: this should work with
```suggestion
    query_str = "j jhep,0903,112"
```

Comment (MJedr) [reply tests/test_elastic_search_visitor.py:2606]: But how can it get the year if it wouldn't be passed? Should the field with publication year be null in query then?

File: inspire_query_parser/visitors/elastic_search_visitor.py [hunk 5]
```diff
@@ -518,6 +520,7 @@ def _preprocess_journal_query_value(self, third_journal_field, old_publication_i
             self.JOURNAL_TITLE,
             self.JOURNAL_VOLUME,
             third_journal_field,
+            self.JOURNAL_YEAR
```
Comment (michamos) [inline inspire_query_parser/visitors/elastic_search_visitor.py:523]: I don't think this is needed, we don't want the year to be passed explicitly as an input to `convert_old_publication_info_to_new`, rather it's an output for some journals.

File: tests/test_elastic_search_visitor.py (hunk 4 above)
Comment (michamos) [reply tests/test_elastic_search_visitor.py:2606]:
```

**Comment by monaawi** [comment]
Don't forget to also bump the query parser in hep

**Pull request merged**